examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+std::vector<std::pair<TString, TString>> GetLeafNames(TTree*);
+
+Bool_t CheckEntriesIdentical(TH1* compHist, std::stringstream& outstream);
+Bool_t CheckHistogramIdentical(TH1* origHist, TH1* newHist, std::stringstream& outstream);
+Bool_t CheckHistogramKolmogorov(TH1* origHist, TH1* newHist, std::stringstream& outstream);
+
+int TreeCompareAuto(TString fileName1 = "", TString fileName2 = "")
+{
+    if (fileName1.IsNull() || fileName2.IsNull()) {
+        cout << "Filenames are not defined" << endl;
+        exit(42);
+    }
+
+    // Get the output tree from the original file
+    TFile* file1 = TFile::Open(fileName1, "READ");
+    if (nullptr == file1)
+        return 42;
+    TTree* tree1 = (TTree*)file1->Get("cbmsim");
+
+    // Get the output tree from the file whcih should be compared
+    TFile* file2 = TFile::Open(fileName2, "READ");
+    if (nullptr == file2)
+        return 42;
+    TTree* tree2 = (TTree*)file2->Get("cbmsim");
+
+    // Add the output tree from the file to compare as friend to the tree
+    // of the original file. This allows to access a data member of the
+    // original file by e.g. StsHit.fX and the data element of the second tree
+    // by tree2.StsHit.fX
+    tree1->AddFriend(tree2, "tree2");
+
+    // Define pairs of data members to compare. One from each tree.
+    // This allows to compare them also if names or the structure of
+    // the classses change.
+    std::vector<std::pair<TString, TString>> leaves = GetLeafNames(tree1);
+
+    std::stringstream outstream;
+    Bool_t okay{kTRUE};
+    Int_t numTestedLeaves{0};
+    Int_t numEmptyLeaves{0};
+    Int_t numLeaves{0};
+    Int_t numFailedLeaves{0};
+    Int_t numIdenticalEntries{0};
+    Int_t numIdenticalHistograms{0};
+    Int_t numKolmogorovHistograms{0};
+
+    for (auto leaf : leaves) {
+        TString leafName = leaf.first;
+        TString leafName1 = leaf.second;
+
+        TString command1 = leafName + ">>htemp";
+        tree1->Draw(command1);
+        int entries1{0};
+        float low1{0.};
+        float high1{0.};
+        int nBins1{0};
+        auto htemp = (TH1F*)gPad->GetPrimitive("htemp");
+        if (htemp) {
+            entries1 = htemp->GetEntries();
+            nBins1 = htemp->GetNbinsX();
+            low1 = htemp->GetXaxis()->GetXmin();
+            high1 = htemp->GetXaxis()->GetXmax();
+        }
+
+        command1 = leafName1 + ">>hist1(" + nBins1 + ", " + low1 + ", " + high1 + ")";
+        //    cout << command1 << endl;
+        tree1->Draw(command1);
+        auto hist1 = (TH1F*)gPad->GetPrimitive("hist1");
+        int entries2{0};
+        float low2{0.};
+        float high2{0.};
+        int nBins2{0};
+        if (hist1) {
+            entries2 = hist1->GetEntries();
+            nBins2 = hist1->GetNbinsX();
+            low2 = hist1->GetXaxis()->GetXmin();
+            high2 = hist1->GetXaxis()->GetXmax();
```

Comment:
data from hist**1** ends up in variable**2**?

Reply:
fixed